client/src/components/Layout/SearchAvailableRooms/SearchAvailableRooms.js

```diff
@@ -0,0 +1,84 @@
+import React, { Component } from 'react'
+import DatePicker from 'react-datepicker'
+import moment from 'moment'
+import axios from 'axios'
+
+import searchIcon from '../../../Images/search.png'
+import calenderIcon from '../../../Images/calendar_icon.png'
+import clockIcon from '../../../Images/clock_icon.png'
+import AvailableRooms from '../AvailableRooms/AvailableRooms'
+
+import './SearchAvailableRooms.css'
+import 'react-datepicker/dist/react-datepicker.css'
+
+class SearchAvailableRooms extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      roomId: [],
+      startDate: new Date(),
+      display: false,
+      availableRooms: [],
+    }
+  }
+
+  handleChange = (date) => {
+    this.setState({
+      startDate: date
+    })
+  }
+
+  searchAvailableRooms = () => {  
+    const date = moment(this.state.startDate.toLocaleString()).format('YYYY-MM-DD h:mm:ss')
+    axios
+      .get(`/api/available-rooms/${date}`)
+      .then((res) => this.setState({ roomId: res.data })).then(() => {
+        this.setState({
+          availableRooms: [],
+        })
+        this.filterRooms()
+      }
+      )
```

Comment:
there are several problems here:

- there is no reason to split this into multiple `then`s. 
- `roomId` is a very misleading name! It indicates just one specific room ID, I had to check the query and took me a while to understand that that's actually an array, the name should indicate that: `rooms`, `roomIds`, `roomArr`, etc. would be good 
- the way you update `availableRooms` is problematic. You first set the `roomId` array plus an empty array for `availableRooms` to the state and then in `filterRooms` you modify this array directly! Modifying the array like that will avoid triggering the re-render because you're not calling `setState`! You then call `setState` to update the `display` variable (so the component gets re-rendered in the end), however that can be avoided altogether. Also, the `filterRooms` method is stateful (it depends on the component's state), you should strive to make things stateless as much as possible - in case of `filterRooms`, it should simply take data through input parameters and return output, no interaction with outside state needed.
- speaking of `filterRooms`, I think I finally understand what that's for. You pass all of the rooms retrieved in the `Rooms` component as a property to this component and then match it with the result of this API call. That's a very complicated and unnecessary way of doing it, all of this should be handled by the query, pls see my comment there. 

All that you should need to do here is:
```js
.then(res => this.setState({
  availableRooms: res.data
}))
```

Reply:
First of all thank you so much for your helpful comments, you can now see my updates and get me your review about it.

Reply:
👍 awesome thanks!